### PR TITLE
Added channels to the index, added channel filtering, cleaned doc

### PR DESCRIPTION
**Pull request by filipnyquist**
This PR adds channels into the index, this combined with the added channel filtering parameter allows us to search specific channels for content.
While adding the code for filtering I split the search query into individual parts so it easier can be edited and read.
The documentation has been changed to swagger, which now means we follow the OpenAPI spec.
We now use the swagger spec to generate API documentation with the help of the `yarn run gendoc` command, which gets the latest API documentation from in line and adds it to the swagger file located in ./docs/swagger.json

This PR has been tested and is working locally. I still want someone else to test it to be sure that the indexing is working as it should.

### CODE REVIEW

**Comment by seanyesmunt** [comment]
@kauffj Now we can add a search bar on the channel page to search for content specific to a channel

**Comment by seanyesmunt** [comment]
https://github.com/lbryio/lbry-desktop/issues/1735

**Comment by tiger5226** [comment]
I will test this soon

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "test": "npm run lint && npm run mocha",
     "prod": "npm run build && node dist/",
     "mocha": "./node_modules/.bin/mocha --compilers js:babel-register --require babel-polyfill",
-    "gendoc": "apidoc -i server/ -o docs/"
```
Comment (tiger5226) [inline package.json]: nothing else is removed from here?

File: server/controllers/lighthouse.js [hunk 2]
```diff
@@ -34,122 +34,142 @@ function getResults (input) {
...
+            },
+          ],
+        },
+      },
+    },
+  };
+  // End of search parts
+  return eclient.search({
+    index  : 'claims',
+    _source: ['name', 'channel', 'value', 'claimId'],
```
Comment (tiger5226) [inline server/controllers/lighthouse.js]: why did you add channel to the source? This is not needed.

File: server/controllers/lighthouse.js [hunk 3]
```diff
@@ -34,122 +34,142 @@ function getResults (input) {
...
-    body   : {
+  // Search is split up into different parts, all search parts goes under this line.
+  let channelSearch;
+  if (input.channel !== undefined) { // If we got a channel argument, lets filter out only that channel
+    channelSearch = {
+      'bool': {
+        'must': {
+          'query_string': {
+            'fields': ['channel'],
+            'query' : `@${input.channel}`,
```
Comment (tiger5226) [inline server/controllers/lighthouse.js]: why do we have `@` here? this should come through the api and it comes from chainquery too already in this format. It is actually a requirement of a channel. We could test for it to see if we were actually passed a channel, however, I don't think that is the job of the api anyway. The channel parameter is a text that we match against in the elastic search database.

File: docs/swagger.json [hunk 4]
```diff
@@ -0,0 +1 @@
...":{"get":{"tags":["Lighthouse API"],"description":"Gets the status of lighthouse.","produces":["application/json"],"responses":{"200":{"description":"Returns the current status of lighthouse","schema":{"type":"object","required":["spaceUsed","claimsInIndex","totSearches"],"properties":{"spaceUsed":{"type":"string","description":"The amount of space used for the search db."},"claimsInIndex":{"type":"integer","description":"The amount of claims in the index."},"totSearches":{"type":"integer","description":"The total amount of searches since the start."}}}}}}}},"definitions":{"SearchItem":{"type":"object","required":["name"],"properties":{"name":{"type":"string","description":"The name of the claim.","example":"lbry-testclaim"},"channel":{"type":"string","description":"The name of the channel that claimed this item.","example":"@lbrytestclaim"},"value":{"type":"object","description":"The decoded value of the metadata for the item.","example":"Here is the decoded claimdata/metadata from the claim in an object."}}},"AutocompleteItem":{"type":"string","description":"A matching search string.","example":"LBRY is amazing"}},"responses":{},"parameters":{},"securityDefinitions":{},"tags":[]}
```
Comment (tiger5226) [inline docs/swagger.json]: so plugging this into the swagger.io editor there are 12 errors. Can you take a look at those? I don't think this is a showstopper.

**Comment by tiger5226** [comment]
@filipnyquist Can we make this 2 pull requests? the swagger stuff is flooding things and I am not sure if I missed anything important. Also the api docs are not really required for the functioning of the api. So worse case scenario we get the channel parameter and a little later we get the docs for example.

Overall great work man! how long did that swagger file take to create!? I know those things are tedious.

**Comment by filipnyquist** [comment]
The swagger file was a pain in the ass, they take ages to fix. I´ll split this into two PRs. Closing this for now.

**Comment by filipnyquist** [comment]
Moved the channel stuff to #92, doc will be in its own PR soon.
